api/client/push.go

```diff
@@ -30,14 +28,6 @@ func (cli *DockerCli) CmdPush(args ...string) error {
 		return err
 	}
 
-	var tag string
-	switch x := ref.(type) {
-	case reference.Canonical:
-		return errors.New("cannot push a digest reference")
```

Comment:
Looks like it was resolved https://github.com/docker/engine-api/pull/162/files#diff-af7c2f642033c3058dc4c26810d875a4R27